### PR TITLE
[ARTP-234] Loadable component

### PR DESCRIPTION
**Pull request by keshaldesai**
This component handles the loading of each component:

- Displays a loading icon when connecting components to their services
- Displays disconnect icon + message when a component is disconnected
- Calls an optional onMount handler to help sping up services

### CODE REVIEW

File: src/client/src/rt-components/loadable/Loadable.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        <LoadableStyle>
+          <div>
+            <DisconnectIcon width={2.75} height={3} />
+          </div>
+          <div>{message}</div>
+        </LoadableStyle>
+      )
+    }
+
+    return null
```
Comment (bhavesh-desai-scratch) [inline src/client/src/rt-components/loadable/Loadable.tsx]: Can this happen?

File: src/client/src/rt-components/loadable/Loadable.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react'
+import { styled } from 'rt-theme'
+import { ServiceConnectionStatus } from 'rt-types'
+import DisconnectIcon from '../icons/DisconnectIcon'
+
+const LoadableStyle = styled.div`
+  width: 100%;
+  height: 100%;
+  border-radius: 0.1875rem;
```
[the remark this answers is not in the document]
Comment (evanrs) [reply src/client/src/rt-components/loadable/Loadable.tsx]: If we leverage REMs to get mobile coverage, yes.

You can test out the effect of using hard values by changing the `:root` font size.

File: src/client/src/rt-components/loadable/Loadable.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  background-color: ${({ theme }) => theme.component.backgroundColor};
+  color: ${({ theme }) => theme.component.textColor};
+  display: flex;
+  align-items: center;
```
[the remark this answers is not in the document]
Comment (evanrs) [reply src/client/src/rt-components/loadable/Loadable.tsx]: I think this is something CSS covers fairly well — it's straight forward and has no vendor idiosyncrasies to obfuscate.

[the remark this answers is not in the document]
Comment (evanrs) [reply src/client/src/rt-components/loadable/Loadable.tsx]: Where as an internal tool with its own API is liable to pollution and extension in ways unique to each developer.

File: src/client/src/rt-components/loadable/Loadable.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  margin: 0 0.0625rem;
+  height: 100%;
+  width: 0.375rem;
+  display: inline-block;
+
+  animation: stretchdelay 1.2s infinite ease-in-out;
+
+  animation-delay: -${({ delay }) => delay}s;
+
+  @keyframes stretchdelay {
```
Comment (evanrs) [inline src/client/src/rt-components/loadable/Loadable.tsx]: Using the `styled` keyframes tool avoid name collisions — also, it's kinda cool 😛 
https://emotion.sh/docs/keyframes

File: src/client/src/rt-components/loadable/Loadable.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  </Spinner>
+)
+
+export default class Loadable extends Component<Props> {
+  componentDidMount = () => this.props.onMount && this.props.onMount()
+
+  render() {
+    const { status, render, message = 'Disconnected' } = this.props
+
+    if (status === ServiceConnectionStatus.CONNECTED) {
```
Comment (evanrs) [inline src/client/src/rt-components/loadable/Loadable.tsx]: And lo the imperative programmer gave unto their descendants the `switch` statement for the ifs were many and the strings constant.

…

Comment (evanrs) [reply src/client/src/rt-components/loadable/Loadable.tsx]: Maybe use a switch statement?

**Pull request merged**